### PR TITLE
reduce number of heap allocations in cluster info

### PR DESCRIPTION
**Pull request by gnusi**
### Scope & Purpose

Reduce number of heap allocations in ClusterInfo

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Review by kvahed — APPROVED** [review]
LGTM

**Review by Dronplane — APPROVED** [review]
LGTM! Nice catch with initializer lists!

File: arangod/Agency/AgencyComm.cpp [hunk 1]
```diff
@@ -542,7 +542,7 @@ void AgencyCommHelper::initialize(std::string const& prefix) {
 
 void AgencyCommHelper::shutdown() {}
 
-std::string AgencyCommHelper::path() { return PREFIX; }
+const std::string& AgencyCommHelper::path() noexcept { return PREFIX; }
```
Comment (jsteemann) [inline arangod/Agency/AgencyComm.cpp:545]: This is obviously valid, but in 99% of the rest of the code we are doing:
```suggestion
std::string const& AgencyCommHelper::path() noexcept { return PREFIX; }
```

File: arangod/Agency/AgencyComm.h [hunk 2]
```diff
@@ -189,7 +189,7 @@ class AgencyCommHelper {
   static void initialize(std::string const& prefix);
   static void shutdown();
 
-  static std::string path();
+  static const std::string& path() noexcept;
```
Comment (jsteemann) [inline arangod/Agency/AgencyComm.h:192]: Same:
```suggestion
  static std::string const& path() noexcept;
```

File: arangod/Cluster/ClusterInfo.cpp [hunk 3]
```diff
@@ -1024,7 +1026,7 @@ void ClusterInfo::loadPlan() {
     }
     auto const& databaseName = database.first;
     _newPlannedViews.erase(databaseName);   // clear views for this database
-    std::vector<std::string> viewsPath {
+    std::initializer_list<std::string_view> viewsPath {
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.cpp:1029]:
```suggestion
    std::initializer_list<std::string_view> const viewsPath {
```

**Review by jsteemann — APPROVED** [review]
LGTM!

**Pull request merged**